src/MetaCompilation.Analyzers/UnitTests/UnitTests.cs

```diff
@@ -1,33 +1,34 @@
 ﻿//  Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-/*
-Unit tests disabled with https://github.com/dotnet/roslyn-analyzers/issues/1307
-
+using System.Threading.Tasks;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CodeFixes;
 using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Testing;
 using Test.Utilities;
 using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<MetaCompilation.Analyzers.MetaCompilationAnalyzer, MetaCompilation.Analyzers.MetaCompilationCodeFixProvider>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<MetaCompilation.Analyzers.MetaCompilationAnalyzer, MetaCompilation.Analyzers.MetaCompilationCodeFixProvider>;
 
 namespace MetaCompilation.Analyzers.UnitTests
 {
-    public class UnitTest : CodeFixVerifier
+    public class UnitTest
     {
         private const string s_messagePrefix = "T: ";
 
         #region default no diagnostics tests
         // no diagnostics
         [Fact]
-        public void TestMethod1()
+        public async Task TestMethod1()
```

Comment:
📝 The tests are converted as follows:

1. Make the test method asynchronous.
2. Delete the previous `expected` definition and call to `VerifyCSharpDiagnostic`. Make a note of the diagnostic ID field used for creating `expected`, as it will be needed below.
3. Replace the call to `VerifyCSharpFix` with the following:

    ```csharp
    await new VerifyCS.Test
    {
        TestState =
        {
            Sources = { test },
            ExpectedDiagnostics =
            {
            },
        },
        FixedState =
        {
            Sources = { fixtest },
            ExpectedDiagnostics =
            {
            },
        },
        CodeFixEquivalenceKey = null,
    }.RunAsync();
    ```

4. Locate the code fix registration in **CodeFixProvider.cs** for the diagnostic seen in step 2, and find the equivalence key string. Copy that string and paste it over the `null` used in the placeholder from step 3.
5. Run the test to get the expected diagnostics for the test state. Copy the block from the test output to the `ExpectedDiagnostics` initializer, and replace the identifier for the diagnostic ID with the one seen in step 2.
6. Run the test again to get the expected diagnostics for the fixed state. Copy the block from the test output to the `ExpectedDiagnostics` initializer. If any diagnostics are reported by the analyzer (as opposed to compilation errors), look up the corresponding field in **DiagnosticAnalyzer.cs** to find the correct identifier.

I converted one `#region` at a time, since step 2 uses the same identifier for all tests within each region.